### PR TITLE
Run non-null-safe tests with Dart 2

### PR DESCRIPTION
**Pull request by kevmoo**
Move the non-null-safe usage of the API to another file, test the file with Dart 2.x.

### CODE REVIEW

File: protoc_plugin/test/generated_message_test.dart [hunk 1]
```diff
@@ -807,12 +720,6 @@ void main() {
     message.m3M = p3.M_M();
   });
 
-  test('testToplevel', () {
-    var message = t.M();
-    message.t = T();
-    t.SApi(null);
```
Comment (kevmoo) [inline protoc_plugin/test/generated_message_test.dart:813]: The call to `SApi` is not valid with null. Not sure what else to do other than delete this. 🤷

**Comment by kevmoo** [comment]
Feel free to squash/merge and delete the branch if you're happy

**Comment by mraleph** [comment]
@osa1 I think this might be related to what you looked at today.

**Comment by osa1** [comment]
@kevmoo we've been discussing these tests in Dart Ecosystem chat. I don't think we want to remove these tests yet as internally we have non-null-safe Dart that uses this library.

**Comment by kevmoo** [comment]
@osa1 – but we can't run Dart 3.0.0- tests against this package w/ them there.

We could pull those out into another test file and only run them on stable or something 🤷

**Comment by osa1** [comment]
Right, I think we will need to move these to another file/package, and configure mono_repo to only test that package/file with SDK 2.x.

Apparently `dart:test` has a separate packages for legacy tests, for example https://github.com/dart-lang/test/blob/master/legacy_tests/spawn_hybrid_with_optout/pubspec.yaml. We could do the same. If we can specify running a specific file with a specific SDK in mono_repo.yaml we could also just move the file outside of `test/` (maybe to `legacy_test/`), that would be simpler.

**Comment by kevmoo** [comment]
@osa1 – sure? Just trying to get CI to green so we can keep the signals of knowing when the SDK breaks things.

**Comment by osa1** [comment]
It seems like on CI `dart analyze` analyzes the new file. I don't understand why that happens, locally when I run `dart analyze` with Dart 3.0.x it doesn't analyze the new file.

We need to somehow make `dart analyze` skip the file with dev SDK.

File: protoc_plugin/test/generated_message_test.dart (hunk 1 above)
Comment (osa1) [reply protoc_plugin/test/generated_message_test.dart:813]: Test moved to legacy test file.

**Review by osa1 — APPROVED** [review]
LGTM now, but someone should review my mono_repo changes :-)

**Review by devoncarew — APPROVED** [review]
mono_repo changes and repo reconfiguration lgtm

**Pull request merged**